src/utils/queuedTxsActions.ts

```diff
@@ -0,0 +1,43 @@
+import { isSignableBy, isTransactionListItem } from '@/utils/transaction-guards'
+import type { TransactionListPage } from '@safe-global/safe-gateway-typescript-sdk'
+
+export type SafeTxsActions = {
+  queued?: string | number
+  signing?: string | number
+}
+
+export type PendingActionsByChain = {
+  [chainId: string]: { [safeAddress: string]: SafeTxsActions }
+}
+
+export const addPendingActionsByChain = (
```

Comment:
I already commented on the naming of this function.

It's just a getter, it shouldn't be called "add".

It's not a util either, it's used only in that hook and should be moved there.